src/main/java/com/three_tech_solutions/slot_app/mappers/StudentMapper.java

```diff
@@ -36,5 +38,34 @@ public StudentResponse toStudentResponse(Student student) {
         return studentDTO;
     }
 
+    public StudentDetailsResponse toStudentDetailsResponse(Student student, List<Payment> payments) {
+        List<PaymentDetailsResponse> paymentsDetailsResponses = payments.stream()
+                .map(this::toPaymentDetailsResponse)
+                .collect(Collectors.toList());
+
+        return new StudentDetailsResponse(
+                student.getId(),
+                student.getName(),
+                student.getLastname(),
+                student.getPhoneNumber(),
+                student.getBirthday(),
+                student.getPathologies(),
+                student.getAdmissionDate(),
+                student.getPlan().getPlanType().name(),
+                student.getPlan().getClassesPerWeek(),
+                student.getPlan().getPaymentDay(),
+                paymentsDetailsResponses
+        );
+    }
```

Comment:
Acá en vez de recibir por parametro los pagos, podemos utilizar directamente la lista de pagos que tiene el estudiante que recibe por parametro, que es la misma


```suggestion
    public StudentDetailsResponse toStudentDetailsResponse(Student student) {
    
        return new StudentDetailsResponse(
                student.getId(),
                student.getName(),
                student.getLastname(),
                student.getPhoneNumber(),
                student.getBirthday(),
                student.getPathologies(),
                student.getAdmissionDate(),
                student.getPlan().getPlanType().name(),
                student.getPlan().getClassesPerWeek(),
                student.getPlan().getPaymentDay(),
                student.getPayments().stream().map(this::toPaymentDetailsResponse).collect(Collectors.toList())
        );
    }
```